erts/emulator/beam/erl_debugger.c

```diff
@@ -550,36 +717,29 @@ stack_frame_fun_info(Process *c_p, ErtsCodePtr pc, Process *rp, int is_return_ad
 static Eterm
 make_value_or_too_large_tuple(Process *p, Eterm val, Uint max_size) {
     Uint val_size;
-    int tup_arity = 2;
+    const int tup2_sz = 3;
     Eterm result, *hp;
 
-    hp = HAlloc(p, tup_arity + 1);
-    result = make_tuple(hp);
-    *hp++ = make_arityval(tup_arity);
-
+    hp = HAlloc(p, tup2_sz);
     val_size = size_object(val);
     if (val_size <= max_size) {
-        *hp++ = am_value;
-        *hp++ = copy_object(val, p);
+        result = TUPLE2(hp, am_value, copy_object(val, p));
     } else {
-        *hp++ = ERTS_MAKE_AM("too_large");
-        *hp++ = make_small(val_size);
+        result = TUPLE2(hp, ERTS_MAKE_AM("too_large"), make_small(val_size));
```

Comment:
Nitpick: maybe `am_system_limit`? Users are already familiar with that terminology.

Reply:
Notice that this is not a system limit: the limit is specified by the caller. The motivation is to give the caller some control on whether they will hit their heap limit by copying a gigantic term while peeking, or avoid the serialization overhead, etc.

Reply:
How about `size` then -- you either return a `{value, Value}` or a `{size, Size}`?